src/plugins/data_views/public/services/has_data.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { CoreStart, HttpStart } from 'kibana/public';
+import { FLEET_ASSETS_TO_IGNORE } from '../../common';
+import { HasDataViewsResponse, IndicesResponse, IndicesResponseModified } from '../';
+
+export class HasData {
+  private removeAliases = (source: IndicesResponseModified): boolean => !source.item.indices;
+
+  private isUserDataIndex = (source: IndicesResponseModified): boolean => {
+    // filter out indices that start with `.`
+    if (source.name.startsWith('.')) return false;
+
+    // filter out sources from FLEET_ASSETS_TO_IGNORE
+    for (const key in FLEET_ASSETS_TO_IGNORE) {
+      if (source.name === (FLEET_ASSETS_TO_IGNORE as any)[key]) return false;
+    }
+
+    // filter out empty sources created by apm server
+    if (source.name.startsWith('apm-')) return false;
+
+    return true;
+  };
+
+  start(core: CoreStart) {
+    const { http } = core;
+    return {
+      /**
+       * Check to see if ES data exists
+       */
+      hasESData: async (): Promise<boolean> => {
+        const hasLocalESData = await this.checkLocalESData(http);
+        if (!hasLocalESData) {
+          const hasRemoteESData = await this.checkRemoteESData(http);
+          return hasRemoteESData;
+        }
+        return hasLocalESData;
+      },
+      /**
+       * Check to see if any data view exists
+       */
+      hasDataView: async (): Promise<boolean> => {
+        const dataViewsCheck = await this.findDataViews(http);
+        return dataViewsCheck;
+      },
+      /**
+       * Check to see if user created data views exist
+       */
+      hasUserDataView: async (): Promise<boolean> => {
+        const userDataViewsCheck = await this.findUserDataViews(http);
+        return userDataViewsCheck;
+      },
+    };
+  }
+
+  // ES Data
+
+  private responseToItemArray = (response: IndicesResponse): IndicesResponseModified[] => {
+    const { indices = [], aliases = [] } = response;
+    const source: IndicesResponseModified[] = [];
+
+    [...indices, ...aliases, ...(response.data_streams || [])].forEach((item) => {
+      source.push({
+        name: item.name,
+        item,
+      });
+    });
+
+    return source;
+  };
+
+  private getIndices = async ({
+    http,
+    pattern,
+    showAllIndices,
+  }: {
+    http: HttpStart;
+    pattern: string;
+    showAllIndices: boolean;
+  }): Promise<IndicesResponseModified[]> =>
+    http
+      .get<IndicesResponse>(`/internal/index-pattern-management/resolve_index/${pattern}`, {
+        query: showAllIndices ? { expand_wildcards: 'all' } : undefined,
+      })
+      .then((response) => {
```

Comment:
Should there be some error catching here, in case the request fails?

Reply:
done